core/src/main/java/com/sequenceiq/cloudbreak/service/LoadBalancerConfigService.java

```diff
@@ -389,6 +445,35 @@ private boolean isEndpointGatewayEnabled(EnvironmentNetworkResponse network) {
         return Optional.ofNullable(knoxTargetGroup);
     }
 
+    private Optional<TargetGroup> setupOozieHATargetGroup(Stack stack, boolean dryRun) {
+        Optional<InstanceGroup> oozieInstanceGroup = getOozieInstanceGroup(stack);
```

Comment:
Can this span across hostGroups? - similar to the way Knox can. Otherwise this ends up restricting Oozie to always be in the same host group, which becomes an unsaid limitation on the definition of custom templates.

Reply:
In AWS yes, but in Azure no in this patch. I am using the availability set as an abstraction similar to target groups in AWS. This simplified the Azure changes. While we can remove this restriction to open up for Oozie in multiple host groups by doing some more code changes in this patch, it will make the Azure Freemarker template code changes and corresponding java code changes, even more, harder to follow.

A little bit of history. Before this patch, it was no for Azure. Because we were using basic SKU and it does not support directly registering instances to the Azure load-balancer. In this patch we changed the SKU to register more than one availability set. Instead we could have chosen to register the instances directly without using any availability set. It was harder to code that way in the Azure Freemarker template.

Reply:
OK. So realistically, Knox would need to be on a single hostGroup as well, correct? (Before and after this patch). The code however is written to allow it to span hostGroups.

Reply:
Knox is also restricted to only one host group before this patch for Azure. In AWS, you are correct, it was allowed in multiple host groups while technically only one host group will work.